src/wpanctl/tool-updateprop.c

```diff
@@ -0,0 +1,242 @@
+			optind++;
+		}
+	}
+
+	if (optind < argc) {
+		fprintf(stderr,
+			"%s: error: Unexpected extra argument: \"%s\"\n",
+			argv[0], argv[optind]);
+		ret = ERRORCODE_BADARG;
+		goto bail;
+	}
+
+	if (!property_name) {
+		fprintf(stderr, "%s: error: Missing property name.\n", argv[0]);
+		ret = ERRORCODE_BADARG;
+		goto bail;
+	}
+
+	if (!property_value) {
+		fprintf(stderr, "%s: error: Missing property value.\n", argv[0]);
+		ret = ERRORCODE_BADARG;
+		goto bail;
+	}
+
+	if (gInterfaceName[0] == 0) {
+		fprintf(stderr,
+		        "%s: error: No WPAN interface set (use the `cd` command, or the `-I` argument for `wpanctl`).\n",
+		        argv[0]);
+		ret = ERRORCODE_BADARG;
+		goto bail;
+	}
+
+	connection = dbus_bus_get(DBUS_BUS_STARTER, &error);
+
+	if (!connection) {
+		dbus_error_free(&error);
+		dbus_error_init(&error);
+		connection = dbus_bus_get(DBUS_BUS_SYSTEM, &error);
+	}
+
+	require_string(connection != NULL, bail, error.message);
+
+	{
+		DBusMessageIter iter;
+		DBusMessageIter list_iter;
+		char path[DBUS_MAXIMUM_NAME_LENGTH+1];
+		char interface_dbus_name[DBUS_MAXIMUM_NAME_LENGTH+1];
+		ret = lookup_dbus_name_from_interface(interface_dbus_name, gInterfaceName);
+		if (ret != 0) {
+			print_error_diagnosis(ret);
+			goto bail;
+		}
+		snprintf(path,
+		         sizeof(path),
+		         "%s/%s",
+		         WPANTUND_DBUS_PATH,
+		         gInterfaceName);
+
+		message = dbus_message_new_method_call(
+		    interface_dbus_name,
+		    path,
+		    WPANTUND_DBUS_APIv1_INTERFACE,
+		    dbus_method_name
+		    );
+
+		dbus_message_append_args(
+		    message,
+		    DBUS_TYPE_STRING, &property_name,
+		    DBUS_TYPE_INVALID
+		    );
+
+		if (property_type == kPropertyType_String) {
+			dbus_message_append_args(
+			    message,
+			    DBUS_TYPE_STRING, &property_value,
+			    DBUS_TYPE_INVALID
+			    );
+		} else if (property_type == kPropertyType_Data) {
+			int length = parse_string_into_data((uint8_t*)property_value,
+			                                    strlen(property_value),
+			                                    property_value);
+			dbus_message_append_args(
+			    message,
+			    DBUS_TYPE_ARRAY, DBUS_TYPE_BYTE, &property_value, length,
+			    DBUS_TYPE_INVALID
+			    );
+		} else {
+			fprintf(stderr, "%s: error: Bad property type\n", argv[0]);
+			ret = ERRORCODE_UNKNOWN;
+			goto bail;
+		}
+
+		reply = dbus_connection_send_with_reply_and_block(
+		    connection,
+		    message,
+		    timeout,
+		    &error
+		    );
+
+		if (!reply) {
+			fprintf(stderr, "%s: error: %s\n", argv[0], error.message);
+			ret = ERRORCODE_TIMEOUT;
+			goto bail;
+		}
+
+		dbus_message_get_args(reply, &error,
+		                      DBUS_TYPE_INT32, &ret,
+		                      DBUS_TYPE_INVALID
+		                      );
+		if (ret)
+			fprintf(stderr, "%s failed with error %d. %s\n", argv[0], ret, wpantund_status_to_cstr(ret));
+	}
+
+bail:
+
+	if (connection)
+		dbus_connection_unref(connection);
```

Comment:
Missing full curly braces. In response to the `goto fail` bug, all new conditional code now needs full curly braces.